### DISCUSSION TITLE
adobe illustrator .ai files in pimcore assets

### DISCUSSION THREAD

**Discussion by lpheller** [pimcore/pimcore#10445] [Q&A / Support]
Hi pimcore community,

I'm currently experiencing some issues when trying to upload adobe illustrator `.ai` files to pimcore assets on a ubuntu production server.

The add-asset request/action is throwing the following exception:
```
stream_get_meta_data(): Argument #1 ($stream) must be of type resource, null given
Exception
TypeError
in vendor/pimcore/pimcore/lib/Helper/TemporaryFileHelperTrait.php (line 36)
```
Uploading the same file to a local testing pimcore environment running on my mac works fine however, as my local environment doesn't have the `imagick` module installed. The server however has and to my knowledge that is desired as it mainly brings advantages over `gd`.

I could get uploading .ai files to work on the production server as well, by setting `low_quality_image_preview` -> `enabled` to `false` within `config/config.yaml`.  However this feels like a workaround, not like a proper solution. 

Is there anyone here who has had a similar problem and has a tip for me?

--- 
Edit
Pimcore Version: 10.1.4

Server: 
- Ubuntu 20.04.2 LTS
- PHP 8.0.7
- all PHP modules listed in Pimcore requirements are installed + imagick

**Comment by dvesh3** [comment]
Pimcore version please? it works for me on latest version 10.1.4

**Reply by lpheller** [reply]
Pimcore was running on v10.1.3 when i opened this post, but I just updated to 10.1.4 to check for any changes. But still not working so it's probably not that.

**Reply by brusch** [reply]
Can you try it on our public demo instance? 
https://github.com/pimcore/pimcore#demo-community-edition

**Reply by lpheller** [reply]
@brush It works! So it has to be something about my server configuration, as I already guessed.
But as I installed all the required modules, do you have any hints on what this could be I'm currently missing?

**Reply by brusch** [reply]
Is .AI in the list of supported formats of Imagick? You can check that in `phpinfo()`.

**Reply by lpheller** [reply]
Yes it is.

<img width="1026" alt="Bildschirmfoto 2021-10-05 um 11 52 24" src="https://user-images.githubusercontent.com/36259611/136001450-5a2f5455-7178-468f-952e-abd42057d8b9.png">

But I only now noticed, that my automatically deployed server doesn't use ImageMagick 7, which your demo version does. So I have to check if I'm able to upgrade this and if it makes any differences.

**Reply by rushitkhunt1996** [reply]
hi, we installed Imagick 7, but still, we can not able to see the preview of .ai files on PIM 10
![image](https://github.com/pimcore/pimcore/assets/60934941/1e0894c7-d087-44f4-a5e4-5a8c78855aba)
